specification/resources/DeploymentStacks.Management/main.tsp

```diff
@@ -0,0 +1,44 @@
+/**
+ * PLEASE DO NOT REMOVE - USED FOR CONVERTER METRICS
+ * Generated by package: @autorest/openapi-to-typespec
+ * Parameters used:
+ *   isFullCompatible: false
+ *   guessResourceKey: false
+ * Version: 0.11.1
+ * Date: 2025-06-06T20:26:04.184Z
+ */
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./models.tsp";
+import "./openApiOverwrites.tsp";
+
+using Azure.ResourceManager;
+using TypeSpec.Versioning;
+
+/**
+ * DeploymentStacks Client
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/missing-operations-endpoint" "Operations API defined separately - namespace is shared between multiple services."
+@armProviderNamespace("Microsoft.Resources")
+@service(#{ title: "DeploymentStacksClient" })
+@versioned(Microsoft.Resources.DeploymentStacks.Versions)
+@armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v5)
+namespace Microsoft.Resources.DeploymentStacks;
+
+/**
+ * The available API versions.
+ */
+enum Versions {
```

Comment:
Do you need to also reference the preview version here?

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/deploymentStacks/preview/2022-08-01-preview/deploymentStacks.json